In [1]:
import pandas as pd

In [2]:
df = pd.read_csv("HealthConnect_Appointment_Data.csv")

In [3]:
df.head()

,appointment_id,patient_id,gender,age,age_group,appointment_type,booking_date,appointment_date,appointment_day,appointment_time,booking_lead_days,previous_appointments,previous_no_shows,reminder_sent,reminder_channel,distance_to_clinic_km,waiting_time_minutes,appointment_outcome
0,HC-00001,P-1613,Female,39,35-44,Follow-up,2/6/2025,2/18/2025,Tuesday,Afternoon,12,2,0,Yes,WhatsApp,19.3,29.0,No-Show
1,HC-00002,P-0813,Male,31,25-34,Specialist Consultation,2/25/2026,2/27/2026,Friday,Morning,2,6,0,Yes,SMS,14.3,42.0,Attended
2,HC-00003,P-1366,Female,50,45-54,General Consultation,11/16/2025,12/24/2025,Wednesday,Morning,38,5,1,Yes,SMS,11.4,11.0,No-Show
3,HC-00004,P-1031,Male,59,55-64,Follow-up,7/18/2025,8/28/2025,Thursday,Evening,41,3,1,Yes,SMS,7.4,35.0,Attended
4,HC-00005,P-1458,Female,34,25-34,Follow-up,7/9/2025,8/25/2025,Monday,Afternoon,47,3,1,Yes,Email,5.6,27.0,No-Show


In [4]:
df.head()

,appointment_id,patient_id,gender,age,age_group,appointment_type,booking_date,appointment_date,appointment_day,appointment_time,booking_lead_days,previous_appointments,previous_no_shows,reminder_sent,reminder_channel,distance_to_clinic_km,waiting_time_minutes,appointment_outcome
0,HC-00001,P-1613,Female,39,35-44,Follow-up,2/6/2025,2/18/2025,Tuesday,Afternoon,12,2,0,Yes,WhatsApp,19.3,29.0,No-Show
1,HC-00002,P-0813,Male,31,25-34,Specialist Consultation,2/25/2026,2/27/2026,Friday,Morning,2,6,0,Yes,SMS,14.3,42.0,Attended
2,HC-00003,P-1366,Female,50,45-54,General Consultation,11/16/2025,12/24/2025,Wednesday,Morning,38,5,1,Yes,SMS,11.4,11.0,No-Show
3,HC-00004,P-1031,Male,59,55-64,Follow-up,7/18/2025,8/28/2025,Thursday,Evening,41,3,1,Yes,SMS,7.4,35.0,Attended
4,HC-00005,P-1458,Female,34,25-34,Follow-up,7/9/2025,8/25/2025,Monday,Afternoon,47,3,1,Yes,Email,5.6,27.0,No-Show


In [5]:
df.shape

(5000, 18)

In [6]:
df.columns

Index(['appointment_id', 'patient_id', 'gender', 'age', 'age_group',
       'appointment_type', 'booking_date', 'appointment_date',
       'appointment_day', 'appointment_time', 'booking_lead_days',
       'previous_appointments', 'previous_no_shows', 'reminder_sent',
       'reminder_channel', 'distance_to_clinic_km', 'waiting_time_minutes',
       'appointment_outcome'],
      dtype='str')

In [7]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 5000 entries, 0 to 4999
Data columns (total 18 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   appointment_id         5000 non-null   str    
 1   patient_id             5000 non-null   str    
 2   gender                 5000 non-null   str    
 3   age                    5000 non-null   int64  
 4   age_group              5000 non-null   str    
 5   appointment_type       5000 non-null   str    
 6   booking_date           5000 non-null   str    
 7   appointment_date       5000 non-null   str    
 8   appointment_day        5000 non-null   str    
 9   appointment_time       5000 non-null   str    
 10  booking_lead_days      5000 non-null   int64  
 11  previous_appointments  5000 non-null   int64  
 12  previous_no_shows      5000 non-null   int64  
 13  reminder_sent          5000 non-null   str    
 14  reminder_channel       3634 non-null   str    
 15  distance_to_cli

In [8]:
df.isnull().sum()

appointment_id              0
patient_id                  0
gender                      0
age                         0
age_group                   0
appointment_type            0
booking_date                0
appointment_date            0
appointment_day             0
appointment_time            0
booking_lead_days           0
previous_appointments       0
previous_no_shows           0
reminder_sent               0
reminder_channel         1366
distance_to_clinic_km      90
waiting_time_minutes       60
appointment_outcome         0
dtype: int64

In [9]:
df.duplicated().sum()

np.int64(0)

### Initial Data Quality Observation

The dataset contains 5,000 appointment records and 18 variables. The initial missing-value assessment identified missing values in three variables: Reminder channel (1,366), Distance to clinic (90), and Waiting time (60). All other variables contain no missing values.

The missing values in Reminder channel require contextual interpretation because the Data Dictionary indicates that "None" represents cases where no reminder was sent. Therefore, these records should not automatically be treated as erroneous missing data.

Further investigation and appropriate treatment of missing values will be considered during the data preparation stage.



In [12]:
df.columns.tolist()

['appointment_id',
 'patient_id',
 'gender',
 'age',
 'age_group',
 'appointment_type',
 'booking_date',
 'appointment_date',
 'appointment_day',
 'appointment_time',
 'booking_lead_days',
 'previous_appointments',
 'previous_no_shows',
 'reminder_sent',
 'reminder_channel',
 'distance_to_clinic_km',
 'waiting_time_minutes',
 'appointment_outcome']

In [13]:
df.columns = df.columns.str.strip().str.replace(" ", "_")

df.columns.tolist()

['appointment_id',
 'patient_id',
 'gender',
 'age',
 'age_group',
 'appointment_type',
 'booking_date',
 'appointment_date',
 'appointment_day',
 'appointment_time',
 'booking_lead_days',
 'previous_appointments',
 'previous_no_shows',
 'reminder_sent',
 'reminder_channel',
 'distance_to_clinic_km',
 'waiting_time_minutes',
 'appointment_outcome']

In [14]:
df["appointment_outcome"].value_counts()

appointment_outcome
No-Show      2423
Attended     2314
Cancelled     263
Name: count, dtype: int64

### Appointment Outcome Observation

The appointment outcome distribution shows 2,423 No-show appointments, 2,314 Attended appointments, and 263 Cancelled appointments out of 5,000 records.

No-show is the largest outcome category in the dataset. This supports the relevance of investigating factors and patterns associated with missed appointments in subsequent analysis.


In [15]:
df.dtypes

appointment_id               str
patient_id                   str
gender                       str
age                        int64
age_group                    str
appointment_type             str
booking_date                 str
appointment_date             str
appointment_day              str
appointment_time             str
booking_lead_days          int64
previous_appointments      int64
previous_no_shows          int64
reminder_sent                str
reminder_channel             str
distance_to_clinic_km    float64
waiting_time_minutes     float64
appointment_outcome          str
dtype: object

### Data Type Observation

The initial data type assessment shows that numerical variables such as age, booking lead days, previous appointments, previous no-shows, distance to the clinic, and waiting time are stored using appropriate numerical data types.

Categorical and identification variables are stored as strings.

However, booking_date and appointment_date are currently stored as strings rather than datetime values. This will need to be considered during the data preparation stage because date-based analysis will be required to investigate appointment scheduling and attendance patterns.

The appointment_time field is also stored as a string and will require review during the scheduling analysis.


In [16]:
df["gender"].value_counts()

gender
Female               2488
Male                 2404
Prefer not to say     108
Name: count, dtype: int64

In [17]:
df["appointment_type"].value_counts()

appointment_type
General Consultation       2086
Follow-up                  1421
Specialist Consultation     900
Diagnostic Test             593
Name: count, dtype: int64

In [18]:
df["appointment_day"].value_counts()

appointment_day
Wednesday    737
Sunday       737
Friday       728
Saturday     724
Monday       701
Tuesday      689
Thursday     684
Name: count, dtype: int64

In [19]:
df["reminder_sent"].value_counts()

reminder_sent
Yes    3634
No     1366
Name: count, dtype: int64

In [20]:
df["reminder_channel"].value_counts(dropna=False)

reminder_channel
SMS         2000
NaN         1366
WhatsApp    1101
Email        533
Name: count, dtype: int64

### Categorical Variable Observation

The categorical variable assessment identified expected categories across gender, appointment type, appointment day, reminder status, and reminder channel.

All seven days of the week are represented in the dataset, providing an appropriate basis for subsequent analysis of appointment outcomes by day.

The reminder assessment shows that 3,634 appointments had a reminder sent, while 1,366 did not. The reminder_channel variable contains 1,366 NaN values, which exactly matches the number of appointments where reminder_sent is "No". This suggests that these values represent appointments where no reminder channel was applicable rather than necessarily representing erroneous missing data.

The categorical variables will be further analysed against appointment outcomes during the exploratory analysis stage.

# Week 4 Initial Findings and Analysis Summary

## Key Findings

The initial inspection of the HealthConnect Appointment Dataset identified 5,000 appointment records across 18 variables.

No duplicate records were identified. Missing values were found in three variables: reminder_channel (1,366), distance_to_clinic_km (90), and waiting_time_minutes (60).

The appointment outcome distribution consists of 2,423 No-show appointments, 2,314 Attended appointments, and 263 Cancelled appointments. No-show is therefore the largest appointment outcome category and will be a major focus of the subsequent analysis.

The categorical variable review identified expected categories across gender, appointment type, appointment day, reminder status, and reminder channel. All seven days of the week are represented in the dataset.

The reminder assessment identified 3,634 appointments where a reminder was sent and 1,366 where no reminder was sent. The 1,366 NaN values in reminder_channel correspond exactly to the appointments where reminder_sent is "No". This will be treated as an important data interpretation point during the next stage rather than automatically classifying these records as erroneous missing data.

The data type assessment also identified that booking_date and appointment_date are currently stored as strings. These fields will require conversion to appropriate date formats during the data preparation stage.

## Analytical Direction

The next stage of the project will investigate appointment attendance and no-show patterns in relation to:

- Previous no-show history
- Reminder status and reminder channel
- Appointment type
- Appointment day and time
- Booking lead time
- Distance to the clinic
- Waiting time
- Relevant patient characteristics

The selected potential KPIs will also be calculated and evaluated during the subsequent analysis.

## Week 5 Focus

The proposed Week 5 focus is data preparation and exploratory analysis, including appropriate treatment of data-quality issues, date conversion, calculation of the selected KPIs, and identification of meaningful patterns associated with appointment attendance and no-shows.